frappe/test_runner.py

```diff
@@ -175,6 +175,7 @@ def run_tests_for_module(module, verbose=False, tests=(), profile=False, junit_x
 		for doctype in module.test_dependencies:
 			make_test_records(doctype, verbose=verbose)
 
+	frappe.db.commit()
```

Comment:
Move this in `make_test_records` instead?

Reply:
Then the error will still persist if there are no test records, right?

Reply:
btw this commit broke it: https://github.com/frappe/frappe/commit/459f65ba189ef36e611dc175bb19a2efda35a9dd 

maybe commit can go before starting the transaction.

Reply:
This is a mess...Anyway